### PR TITLE
chore(auth): enable eslint and remove tslint

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

Please see commits message for details.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/providers/cognito/utils/srp/calculate/calculateS.native.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const calculateS = async ({
 	k,
 	x,
 	B,
-	N,
+	N: _,
```
Comment (HuiSF) [inline packages/auth/src/providers/cognito/utils/srp/calculate/calculateS.native.ts:14]: Unused param.

**Review by ashika112 — DISMISSED** [review]

File: packages/auth/package.json [hunk 2]
```diff
@@ -25,7 +25,8 @@
 		"clean": "npm run clean:size && rimraf lib-esm lib dist",
 		"clean:size": "rimraf dual-publish-tmp tmp*",
 		"format": "echo \"Not implemented\"",
-		"lint": "tslint '{src}/**/*.ts' && npm run ts-coverage",
+		"lint": "eslint '**/*.{ts,tsx}' && npm run ts-coverage",
```
Comment (elorzafe) [inline packages/auth/package.json:28]: why not `src`?

Comment (HuiSF) [reply packages/auth/package.json:28]: We need to lint `__tests__` and any other directories that may contain implementation.

File: packages/auth/src/providers/cognito/credentialsProvider/IdentityIdProvider.ts [hunk 3]
```diff
@@ -17,9 +18,9 @@ const logger = new ConsoleLogger('CognitoIdentityIdProvider');
  *
  * @param tokens - The AuthTokens received after SignIn
  * @returns string
- * @throws configuration excpetions: {@link InvalidIdentityPoolIdException }
+ * @throws configuration exceptions: {@link InvalidIdentityPoolIdException }
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/credentialsProvider/IdentityIdProvider.ts:21]: We should probably just remove these links, or replace with `AuthConfigurationErrorCode`. Okay leaving this out of this PR for now I think though.

File: packages/auth/src/providers/cognito/utils/oauth/getRedirectUrl.native.ts [hunk 4]
```diff
@@ -6,8 +6,7 @@ import { invalidRedirectException } from '../../../../errors/constants';
 /** @internal */
 export function getRedirectUrl(redirects: string[]): string {
 	const redirect = redirects?.find(
-		redirect =>
-			!redirect.startsWith('http://') && !redirect.startsWith('https://'),
+		r => !r.startsWith('http://') && !r.startsWith('https://'),
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/utils/oauth/getRedirectUrl.native.ts:9]: Nit, it's not letting me add a suggestion for this but maybe `redirectUrl`?

**Review by jimblanc — APPROVED** [review]

File: packages/auth/src/providers/cognito/credentialsProvider/IdentityIdProvider.ts (hunk 3 above)
Comment (HuiSF) [reply packages/auth/src/providers/cognito/credentialsProvider/IdentityIdProvider.ts:21]: Yes, I'm planning to address all warnings regarding non-existing types across packages once all of them gets migrated to eslint.

**Review by elorzafe — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**